consensus/scheme/rolldpos/rolldpos.go

```diff
@@ -36,25 +36,84 @@ var (
 	ErrNotEnoughCandidates = errors.New("Candidate pool does not have enough candidates")
 )
 
-// ChainManager defines the blockchain interface
-type ChainManager interface {
-	// Genesis returns the genesis
-	Genesis() genesis.Genesis
-	// BlockHeaderByHeight return block header by height
-	BlockHeaderByHeight(height uint64) (*block.Header, error)
-	// BlockFooterByHeight return block footer by height
-	BlockFooterByHeight(height uint64) (*block.Footer, error)
-	// MintNewBlock creates a new block with given actions
-	// Note: the coinbase transfer will be added to the given transfers when minting a new block
-	MintNewBlock(timestamp time.Time) (*block.Block, error)
-	// CommitBlock validates and appends a block to the chain
-	CommitBlock(blk *block.Block) error
-	// ValidateBlock validates a new block before adding it to the blockchain
-	ValidateBlock(blk *block.Block) error
-	// TipHeight returns tip block's height
-	TipHeight() uint64
-	// ChainAddress returns chain address on parent chain, the root chain return empty.
-	ChainAddress() string
+type (
+	// ChainManager defines the blockchain interface
+	ChainManager interface {
+		// BlockProposeTime return propose time by height
+		BlockProposeTime(uint64) (time.Time, error)
+		// BlockCommitTime return commit time by height
+		BlockCommitTime(uint64) (time.Time, error)
+		// GenesisTime return Genesis time by default
+		GenesisTime() time.Time
+		// MintNewBlock creates a new block with given actions
+		// Note: the coinbase transfer will be added to the given transfers when minting a new block
+		MintNewBlock(timestamp time.Time) (*block.Block, error)
+		// CommitBlock validates and appends a block to the chain
+		CommitBlock(blk *block.Block) error
+		// ValidateBlock validates a new block before adding it to the blockchain
+		ValidateBlock(blk *block.Block) error
+		// TipHeight returns tip block's height
+		TipHeight() uint64
+		// ChainAddress returns chain address on parent chain, the root chain return empty.
+		ChainAddress() string
+	}
+	chainManager struct {
+		bc blockchain.Blockchain
+	}
+)
+
+// BlockProposeTime return propose time by height
+func (cm *chainManager) BlockProposeTime(height uint64) (time.Time, error) {
+	header, err := cm.bc.BlockHeaderByHeight(height)
+	if err != nil {
+		return time.Now(), errors.Wrapf(
+			err, "error when getting the block at height: %d",
+			height,
+		)
+	}
+	return header.Timestamp(), nil
+}
+
+// BlockCommitTime return commit time by height
+func (cm *chainManager) BlockCommitTime(height uint64) (time.Time, error) {
+	footer, err := cm.bc.BlockFooterByHeight(height)
+	if err != nil {
+		return time.Now(), errors.Wrapf(
+			err, "error when getting the block at height: %d",
+			height,
+		)
+	}
+	return footer.CommitTime(), nil
+}
+
+// GenesisTime return Genesis time by default
+func (cm *chainManager) GenesisTime() time.Time {
```

Comment:
if both `BlockCommitTime(0)` and `BlockProposeTime(0)` return `bc.Genesis().Timestamp`, we can delete this function from the interface.

Reply:
yes